### PR TITLE
Add basic pre-commit checks

### PR DESCRIPTION
**Pull request by ArturoGuerra**
Adds some basic pre-commit checks and modified required files to pass all added checks:
black
codespell
flake8
isort
mypy

### CODE REVIEW

**Comment by ArturoGuerra** [comment]
Well thats unexpected

**Comment by ArturoGuerra** [comment]
Okay it appears this is due to the fact the current CI pipeline is still using python3.9 instead of python3.10

File: scripts/run-in-env.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env sh
```
Comment (MartinHjelmare) [inline scripts/run-in-env.sh:1]: Do we really need this script? I'd assume that the contributor has the environment active when committing.

Comment (ArturoGuerra) [reply scripts/run-in-env.sh:1]: Probably not only reason I added it was because the HA core repo has it

Comment (MartinHjelmare) [reply scripts/run-in-env.sh:1]: Yeah, I think it was added there when some contributors were running GUI tools that didn't activate the environment automatically. I don't think we need it here as we'll have a smaller contributor community. We can update later if we see a need.

Comment (ArturoGuerra) [reply scripts/run-in-env.sh:1]: Kk I'll remove in the morning 😃

**Comment by ArturoGuerra** [comment]
The only thing missing rn is mypy but we can add that once support for python3.9 is removed

**Pull request merged**